relay-server/src/utils/rate_limits.rs

```diff
@@ -850,4 +900,69 @@ mod tests {
         mock.assert_call(DataCategory::Attachment, None);
         mock.assert_call(DataCategory::Session, None);
     }
+
+    #[test]
+    fn test_enforce_transaction_no_metrics_extracted() {
+        let mut envelope = envelope![Transaction];
+
+        let mut mock = MockLimiter::default().deny(DataCategory::TransactionProcessed);
+        let limiter = EnvelopeLimiter::new(|s, q| mock.check(s, q));
+        let (enforcement, limits) = limiter.enforce(&mut envelope, &scoping()).unwrap();
+
+        assert!(limits.is_limited());
+        assert!(enforcement.extracted_transaction_metrics.is_active());
+        assert!(enforcement.event.is_active());
+        mock.assert_call(DataCategory::TransactionProcessed, Some(0));
+    }
+
+    #[test]
+    fn test_enforce_transaction_metrics_extracted() {
+        let mut envelope = envelope![Transaction];
+        envelope
+            .get_item_by_mut(|item| *item.ty() == ItemType::Transaction)
+            .unwrap()
+            .set_metrics_extracted(true);
+
+        let mut mock = MockLimiter::default().deny(DataCategory::TransactionProcessed);
+        let limiter = EnvelopeLimiter::new(|s, q| mock.check(s, q));
+        let (enforcement, limits) = limiter.enforce(&mut envelope, &scoping()).unwrap();
+
+        assert!(limits.is_limited());
+        assert!(enforcement.extracted_transaction_metrics.is_active());
+        assert!(enforcement.event.is_active());
+    }
+
+    #[test]
+    fn test_enforce_transaction_no_indexing_quota() {
+        let mut envelope = envelope![Transaction];
+
+        let mut mock = MockLimiter::default().deny(DataCategory::Transaction);
+        let limiter = EnvelopeLimiter::new(|s, q| mock.check(s, q));
+        let (enforcement, limits) = limiter.enforce(&mut envelope, &scoping()).unwrap();
+
+        assert!(!limits.is_limited());
+        assert!(!enforcement.extracted_transaction_metrics.is_active());
+
+        // Note we never check for the DataCategory::Transaction so we are not setting this
+        // to enforced.  After metrics extraction will check the rate limiter again and then
+        // this will actually do something.
```

Comment:
I'd like to have a test covering this, no opinions on the type of the test. At this point, I don't feel confident in applying changes to how this whole logic works and expect it to work, so the more coverage we have for these changes the better.

Reply:
We are testing this is the very next test though: `test_enforce_transaction_metrics_extracted_no_indexing_quota`